### DISCUSSION TITLE
FR: native-image metadata

### DISCUSSION THREAD

**Discussion by dcolascione** [sashirestela/simple-openai#289] [Ideas]
It'd be nice if the reflection stuff were somehow included with the simple-openai library so that it's no longer necessary to use the native-image agent to dynamically collect reflection and proxy information.

**Comment by sashirestela** [comment]
@dcolascione Thanks for using simple-openai! I'm curious about what you are doing with that information?

**Comment by dcolascione** [comment]
Python's official OAI module takes 700ms to load. With GraalVM native image, the equivalent Java program with the same logic starts in 10ms.

**Comment by dcolascione** [comment]
That said, my FR should probably get punted over to Lombok. :-( Lombok uses runtime reflection for its various fancy automatic builder classes, and the right way to clue GraalVM native-image into this reflection scheme would be for Lombok to provide a Graal "feature" plugin that scans the code and adds any needed reflection declarations to the native image build configuration.
